tonic/src/codec/compression.rs

```diff
@@ -78,7 +120,7 @@ pub(crate) struct CompressionSettings {
 }
 
 /// The compression encodings Tonic supports.
-#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+#[derive(Clone, Copy, Debug, EnumCountMacro, PartialEq, Eq)]
```

Comment:
I don't think we want to take on the extra dependency just for this. Should be pretty easy to just implement methods by hand to do what we need here?

Reply:
I have changed it to be manually tracked- when you say by hand, do you intend a macro, or just a regular vec?